Adds new Advanced Behavioral Detections section (#4296)
elastic/security-docs: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc

[[behavioral-detection-use-cases]]
= Behavioral detection use cases

> Behavioral use cases detect potential insider and external threats based on observed user actions and host activities. These use cases employ a threat-centric approach to flag deviations or suspicious activities by analyzing patterns, anomalies, and context enrichment. 

Reviewer: ```suggestion
Behavioral detection identifies potential internal and external threats based on user and host activity. It employs a threat-centric approach to flag deviations from the norm and suspicious activity by analyzing patterns, anomalies, and context enrichment. 
```
I'm wondering about the capitalization / proper name for this set of features, since on the Advanced Behavioral Detections page, you capitalized "Advanced Behavioral Detections", but that full name isn't used here. Not sure what's correct for this feature but we should be consistent. In other words I'm wondering if this paragraph should start with "Advanced Behavioral Detections identify..."?

Author: The full feature hierarchy is:
- Advanced Entity Analytics
  - Entity Risk Scoring
  - Advanced Behavioral Detections
    - Anomaly detection
    - Behavioral detection use cases
    
Essentially, anomaly detection and behavioral detection use cases are two capabilities of the Advanced Behavioral Detections feature. In a previous PR where Entity Risk Scoring was first introduced, it was agreed to capitalize it as a feature (similar to how we capitalize AI Assistant, for example). I capitalized Advanced Behavioral Detections since it's at the same hierarchy level as Entity Risk Scoring. I'm not sure if we classify anomaly detection and behavioral detection use cases as features – @SourinPaul, @jmikell821 what are your thoughts?

Reviewer 2: I am leaning towards sentence case for `Advanced behavioral detections` for consistency, unless @SourinPaul has a business need otherwise.

Author: In that case, should we also lowercase `Entity risk scoring`?

Author: To close the loop on this, per the slack convo, we're lowercasing both `advanced behavioral detections` and `entity risk scoring`.

> Elastic builds behavioral detection use cases on its foundational SIEM detection capabilities, leveraging {ml} algorithms and AI to enable proactive threat detection and hunting.

Reviewer: This reads like we're saying Elastic builds use cases, which doesn't sound right to me. We build a feature or capability that _has_ use cases. Also I think its redundant to say machine learning and AI, feels buzzword-y. 
```suggestion
{elastic-sec}'s behavioral detection feature builds on its foundational SIEM detection capabilities, leveraging {ml} algorithms to enable proactive threat detection and hunting.
```

Author: I think you're right, but I'll defer to @SourinPaul to confirm.

Reviewer 2: Agree with this suggestion too ^.

[float]
[[ml-integrations]]
=== Elastic {integrations} for behavioral detection use cases

> Elastic {integrations-docs}[{integrations}] provide a convenient way to enable behavioral detection use case packages. These integrations streamline the deployment of package components, such as data ingestion, transforms, rules, {ml} jobs, and scripts to your environment. 

Reviewer: Reading this from the perspective of a naive user, I'm not clear on what "packages" means. (A software package, or a set of capabilities targeted at a particular use case? Both?). Reading the links below helps clarify, but I think this page could keep it simple and just refer to capabilities. Users could then learn about each package when they click through to a particular integration.

Also, do the integrations "streamline" the deployment of ML packages (capabilities?) or do they "enable" them? (Is there a less streamlined way to enable them that doesn't use the integrations? I'm not sure).

I also suggest taking out the link on "integrations" because we already link to several specific integrations below, and I'm not sure we want to direct the user to the general integrations page from here.
```suggestion
Behavioral detection integrations provide a convenient way to enable behavioral detection capabilities. They streamline the deployment of components that implement behavioral detection, such as data ingestion, transforms, rules, {ml} jobs, and scripts.
```

Author: @SourinPaul is there any difference between a 'package' and 'integration'? If they are the same, then we may need to rephrase this section to avoid confusion.

.Requirements
[sidebar]
--
* Elastic {integrations} require a https://www.elastic.co/pricing[Platinum subscription] or higher.
* To learn more about the requirements for using {ml} jobs, refer to <<ml-requirements, Machine learning job and rule requirements>>.
--

> The following is a list of behavioral detection use cases and their integrations:

Reviewer: ```suggestion
The following is a list of behavioral detection integrations for various use cases:
```
